src/main/scala/gen/gmos/GmosNLongslitD.scala

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) 2019 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package basic
+package gen.gmos
+
+import basic.search.Constraints
+import basic.syntax.all._
+
+import gem.Step
+import gem.config.DynamicConfig.GmosN
+import gem.config.GmosConfig._
+import gem.enum._
+import gem.math.{ MagnitudeValue, Wavelength }
+
+import cats.Functor
+import cats.effect.Sync
+import cats.implicits._
+import fs2.Stream
+import monocle.{Prism, Optional}
+
+import java.time.Duration
+import java.util.concurrent.TimeUnit.NANOSECONDS
+
+import scala.concurrent.duration.FiniteDuration
+
+/**
+ * Dynamic sequence generation for GMOS North Longslit.
+ */
+sealed trait GmosNLongslitD {
+
+  /**
+   * Generates a sequence that is as long as necessary to achieve the desired
+   * s/n ratio.  Integration time is adjusted throughout to match conditions.
+   *
+   * @param itc integration time calculator
+   * @param acquired a program that can be evaluated to determine whether an
+   *                 acquisition (or reacquisition) is necessary
+   * @param reachedS2N a program that can be evaluated to determine whether the
+   *                   desired s/n ratio has been reached
+   * @tparam F effect
+   *
+   * @return a sequence that acquires (and reacquires as necessary) the target
+   *         and collects science data and calibrations until the desired s/n
+   *         ratio is achieved
+   */
+  def sequence[F[_]: Sync](
+    itc:        Itc[F],
+    acquired:   F[Boolean],
+    reachedS2N: F[Boolean]
+  ): Stream[F, Step.GmosN]
+
+}
+
+
+object GmosNLongslitD {
+
+  object exposureTime {
+
+    // Maximum exposure time we can handle
+    val Max: Duration =
```

Comment:
I think GMOS also has a minimum allowed exposure. I may be misremembering.

Reply:
I imagine you're right but I was focused on ensuring that we can convert between `java.time.Duration`, which is currently what we store in the model for exposure time, and `scala.concurrent.duration.FiniteDuration` which is what we have currently have in the `Itc` stub.  `java.time.Duration` can express durations "greater than the current estimated age of the universe" while `FiniteDuration` is limited about 292 years. Both are beyond any realistic maximum exposure time.